implementations/rust/ockam/ockam_command/src/node/create.rs

```
@@ -352,8 +353,46 @@ async fn run_foreground_node(
         }
     }
 
+    // Create a channel for communicating back to the main thread
+    let (tx, mut rx) = tokio::sync::mpsc::channel(2);
+
+    // Register a handler for SIGINT, SIGTERM, SIGHUP
+    let tx_clone = tx.clone();
+    let mut opts_clone = opts.clone();
+    ctrlc::set_handler(move || {
+        tx_clone.blocking_send(()).unwrap();
+        opts_clone
+            .shell
+            .write_line(format!("{} Ctrl+C signal received", "!".light_yellow()).as_str())
+            .unwrap();
```

Same here about ignoring the error.